### PR TITLE
Bump lang/ccl version number and update license.

### PR DESCRIPTION
**Pull request by kakuhen**
#### Description
We update the license to reflect the change made by the developers. [Source](https://lists.clozure.com/pipermail/openmcl-devel/2016-January/011174.html)
We also update the version number and relevant entries so that lang/ccl installs the 1.12.1 release.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H1217 x86_64
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by raimue** [comment]
Looks like the license has been [wrong for a while](https://lists.clozure.com/pipermail/openmcl-devel/2016-January/011174.html). But I don't think this has any consequences as the port has no dependents.

Please follow [CommitMessages](https://trac.macports.org/wiki/CommitMessages) for the commit message format. In this case, please start the commit message with the `ccl: ...` prefix to make it easier to follow the git log.

File: lang/ccl/Portfile [hunk 1]
```diff
@@ -3,15 +3,16 @@
 PortSystem      1.0
 
 PortGroup       github 1.0
-github.setup    Clozure ccl 1.12 v
+github.setup    Clozure ccl 1.12.1 v
 
 name		    ccl
 revision        0
+dist_subdir     ${name}/${version}
```
Comment (reneeotten) [inline lang/ccl/Portfile:10]: why is this needed - it seems like it should work without?

Comment (kakuhen) [reply lang/ccl/Portfile:10]: On macOS Catalina and later, you have to download a "kernel" for CCL, and as far as I am aware, they are all distributed with the name "darwinx86," so there is no way for MacPorts to determine updates to the kernel without this addition. In fact, without this line, you will get checksum errors, as MacPorts will attempt downloading the 1.12 kernel from its own repository, as opposed to the (slightly) different 1.12.1 kernel that is distributed with this release.

Comment (reneeotten) [reply lang/ccl/Portfile:10]: ah, I see upstream uses an unversioned file for bootstrapping and you're following the [unversioned distfiles](https://trac.macports.org/wiki/PortfileRecipes#unversioned-distfiles) recipe. That seems indeed correct!
